### PR TITLE
Enable/disable Clerk on channel

### PR DESCRIPTION
**Pull request by lruozzi9**
This PR will solve issue #3. Any suggestion for naming class is well appreciated.

### CODE REVIEW

File: src/Resources/config/services.xml [hunk 1]
```diff
@@ -73,5 +76,9 @@
         <service id="Webgriffe\SyliusClerkPlugin\Service\PublicApiKeyProvider">
             <argument/>
         </service>
+
+        <service id="webgriffe_sylius_clerk.resolver.channel_enabled" class="Webgriffe\SyliusClerkPlugin\Service\ChannelApiKeyChecker">
```
Comment (mmenozzi) [inline src/Resources/config/services.xml:80]: The class is called "checker" but the service is called "resolver". Given the fact that the interface returns a boolean I prefer the name "checker"

File: src/Service/ChannelApiKeyChecker.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        /** @var array $clerkStore */
+        foreach ($this->clerkStores as $clerkStore) {
+            if ($clerkStore['channel_code'] === $channel->getCode()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```
Comment (mmenozzi) [inline src/Service/ChannelApiKeyChecker.php:28]: Why not using the `PublicApiKeyResolver` and `PrivateApiKeyResolver` instead? This duplicates the logic already implemented in resolvers and the logic implemented in the dependency injection extension.

**Comment by mmenozzi** [comment]
There are also these BC breaks:
```
[BC] CHANGED: The number of required arguments for Webgriffe\SyliusClerkPlugin\Controller\SalesTrackingController#__construct() increased from 2 to 3

[BC] CHANGED: The number of required arguments for Webgriffe\SyliusClerkPlugin\Controller\TrackingCodeController#__construct() increased from 2 to 3

[BC] CHANGED: The number of required arguments for Webgriffe\SyliusClerkPlugin\Controller\FeedController#__construct() increased from 3 to 4
```
They could be easily removed by making new constructor arguments optional.

File: src/Resources/config/services.xml (hunk 1 above)
Comment (lruozzi9) [reply src/Resources/config/services.xml:80]: Whops!

**Review by mmenozzi — APPROVED** [review]
Well done!

**Pull request merged**